boot/kernel_os.go

```diff
@@ -20,80 +20,44 @@
 package boot
 
 import (
-	"errors"
 	"fmt"
 	"path/filepath"
-	"strings"
 
 	"github.com/snapcore/snapd/bootloader"
 	"github.com/snapcore/snapd/logger"
-	"github.com/snapcore/snapd/release"
 	"github.com/snapcore/snapd/snap"
 )
 
-// RemoveKernelAssets removes the unpacked kernel/initrd for the given
-// kernel snap.
-func RemoveKernelAssets(s snap.PlaceInfo) error {
-	bootloader, err := bootloader.Find()
-	if err != nil {
-		return fmt.Errorf("no not remove kernel assets: %s", err)
-	}
-
-	// ask bootloader to remove the kernel assets if needed
-	return bootloader.RemoveKernelAssets(s)
-}
-
-// ExtractKernelAssets extracts kernel/initrd/dtb data from the given
-// kernel snap, if required, to a versioned bootloader directory so
-// that the bootloader can use it.
-func ExtractKernelAssets(s *snap.Info, snapf snap.Container) error {
-	if s.GetType() != snap.TypeKernel {
-		return fmt.Errorf("cannot extract kernel assets from snap type %q", s.GetType())
-	}
-
-	bootloader, err := bootloader.Find()
-	if err != nil {
-		return fmt.Errorf("cannot extract kernel assets: %s", err)
-	}
-
-	// ask bootloader to extract the kernel assets if needed
-	return bootloader.ExtractKernelAssets(s, snapf)
+type coreBootParticipant struct {
+	s snap.PlaceInfo
+	t snap.Type
 }
 
-// SetNextBoot will schedule the given OS or base or kernel snap to be
-// used in the next boot. For base snaps it up to the caller to select
-// the right bootable base (from the model assertion).
-func SetNextBoot(s *snap.Info) error {
-	if release.OnClassic {
-		return fmt.Errorf("cannot set next boot on classic systems")
-	}
-
-	if s.GetType() != snap.TypeOS && s.GetType() != snap.TypeKernel && s.GetType() != snap.TypeBase {
-		return fmt.Errorf("cannot set next boot to snap %q with type %q", s.SnapName(), s.GetType())
-	}
+var _ BootParticipant = (*coreBootParticipant)(nil)
```

Comment:
I understand this is a bit of static assertion but I think it warrants a comment